src/wp-includes/class-wp-block.php

```diff
@@ -333,6 +333,73 @@ private function replace_html( string $block_content, string $attribute_name, $s
 		switch ( $block_type->attributes[ $attribute_name ]['source'] ) {
 			case 'html':
 			case 'rich-text':
+				if ( 'core/image' === $this->name && 'caption' === $attribute_name ) {
+					// Create private anonymous class until the HTML API provides `set_inner_html` method.
+					$bindings_processor = new class( $block_content, WP_HTML_Processor::CONSTRUCTOR_UNLOCK_CODE ) extends WP_HTML_Processor {
+						/**
+						 * Replace the inner text of an HTML with the passed content.
+						 *
+						 * THIS IS A TEMPORARY SOLUTION IN CORE NOT TO BE EMULATED.
+						 * IT IS A TEMPORARY SOLUTION THAT JUST WORKS FOR THIS SPECIFIC
+						 * USE CASE UNTIL THE HTML PROCESSOR PROVIDES ITS OWN METHOD.
+						 *
+						 * @since 6.6.0
+						 *
+						 * @param string $new_content New text to insert in the HTML element.
+						 * @return bool Whether the inner text was properly replaced.
+						 */
+						public function set_inner_text( $new_content ) {
+							/*
+							 * THIS IS A STOP-GAP MEASURE NOT TO BE EMULATED.
+							 *
+							 * Check that the processor is paused on an opener tag.
+							 *
+							 */
+							if (
+								WP_HTML_Processor::STATE_MATCHED_TAG !== $this->parser_state ||
+								$this->is_tag_closer()
+							) {
+								return false;
+							}
```

Comment:
I would rename this to `set_content_between_figcaption_balanced_tags` to discourage users from copying this method, and ensure it only processes it if the tag is a `figcaption`.

```suggestion
						/**
						 * Replace the inner content of a figcaption element with the passed content.
						 *
						 * THIS IS A TEMPORARY SOLUTION IN CORE NOT TO BE EMULATED.
						 * IT IS A TEMPORARY SOLUTION THAT JUST WORKS FOR THIS SPECIFIC
						 * USE CASE UNTIL THE HTML PROCESSOR PROVIDES ITS OWN METHOD.
						 *
						 * @since 6.6.0
						 *
						 * @param string $new_content New content to insert in the figcaption element.
						 * @return bool Whether the inner content was properly replaced.
						 */
						public function set_content_between_figcaption_balanced_tags( $new_content ) {
							/*
							 * THIS IS A STOP-GAP MEASURE NOT TO BE EMULATED.
							 *
							 * Check that the processor is paused on an opener tag.
							 *
							 */
							if (
								WP_HTML_Processor::STATE_MATCHED_TAG !== $this->parser_state ||
                'figcaption' !== $this->get_tag() ||
								$this->is_tag_closer()
							) {
								return false;
							}
```

Reply:
I think we may need in the future this function not only for captions, but also for more attributes 😅

Reply:
Based on the same logic applied in [this comment](https://github.com/WordPress/wordpress-develop/pull/6838#issuecomment-2175880633), where I moved the logic inside the conditional to check it is an image block and caption attribute, I've pushed the commit with the suggestion. We can always revisit the logic if/when using it for more blocks.